### ISSUE TITLE
Path traversal Plugin backup

### ISSUE THREAD

**Issue by t0gu** [bludit/bludit#1267]
### Describe your problem
Hello all, in the backup plugin there is the possibility to recover internal files from the OS..it was observed in the source code, that there is no validation to recover files using ../../../

### Steps to reproduce the problem
1- Install the backup plugin
2- Make a backup
3- Download the backup using an https proxy like burpsuite [https://portswigger.net/burp/communitydownload] and change the value of the file parameter to "../../../../ .. /../../../../../../../../../../../../../etc/passwd "
4- Verify that the / etc / passwd file will be downloaded

==> PoC

![path_traversal_plugin](https://user-images.githubusercontent.com/61463998/93358751-3fb5bd00-f818-11ea-9dd0-33f299331d34.gif)

==> Vulnerable code 
![Selection_003](https://user-images.githubusercontent.com/61463998/93358928-6ffd5b80-f818-11ea-8799-be5172665819.png)

==> Fixing 
```
The most effective way to prevent file path traversal vulnerabilities is to avoid passing user-supplied input to filesystem APIs altogether. Many application functions that do this can be rewritten to deliver the same behavior in a safer way.

If it is considered unavoidable to pass user-supplied input to filesystem APIs, then two layers of defense should be used together to prevent attacks:

The application should validate the user input before processing it. Ideally, the validation should compare against a whitelist of permitted values. If that isn't possible for the required functionality, then the validation should verify that the input contains only permitted content, such as purely alphanumeric characters.
After validating the supplied input, the application should append the input to the base directory and use a platform filesystem API to canonicalize the path. It should verify that the canonicalized path starts with the expected base directory.
Below is an example of some simple Java code to validate the canonical path of a file based on user input:

File file = new File(BASE_DIRECTORY, userInput);
if (file.getCanonicalPath().startsWith(BASE_DIRECTORY)) {
    // process file
}
```
* More info here [https://owasp.org/www-community/attacks/Path_Traversal]
### Bludit version
3.12.0

### Version
Stantard

**Comment by anaggh** [comment]
This was fixed in 3.13.1 https://github.com/bludit/bludit/commit/7689aa5fb40aad107fdc6599cd0a04af1545c060 The glob expression is not user controlled.

Are you able to reproduce this bug in 3.13.1?

**Comment by t0gu** [comment]
@anaggh Ok, i'll reproduce is these version.

**Comment by t0gu** [comment]
Fixed at 3.13.1

**Comment by anaggh** [comment]
Thanks. Please close the issue.
